src/components/SchedulePanel/index.js

```diff
@@ -0,0 +1,361 @@
+import React, { Component, Fragment } from 'react';
+import Button from '@material-ui/core/Button';
+import Switch from '@material-ui/core/Switch';
+import CourseSchedule from './components/CourseSchedule';
+import ExpansionPanel from '@material-ui/core/ExpansionPanel';
+import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
+import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import Typography from '@material-ui/core/Typography';
+import { withStyles } from '@material-ui/core/styles';
+import { gordonColors } from '../../theme';
+import HoursDialog from './components/OfficeHoursDialog';
+import RemoveHoursDialog from './components/RemoveHoursDialog';
+import EditDescriptionDialog from './components/EditDescriptionDialog';
+import TimeAgo from 'react-timeago';
+
+import schedulecontrol from './../../services/schedulecontrol';
+
+import './schedulepanel.css';
+import myschedule from '../../services/myschedule';
+
+import GordonLoader from '../../components/Loader';
+
+const styles = {
+  colorSwitchBase: {
+    color: gordonColors.neutral.lightGray,
+    '&$colorChecked': {
+      color: gordonColors.primary.cyan,
+      '& + $colorBar': {
+        backgroundColor: gordonColors.primary.cyan,
+      },
+    },
+  },
+  colorBar: {},
+  colorChecked: {},
+};
+
+class GordonSchedulePanel extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      myProf: false, //if my profile page
+      isSchedulePrivate: 0,
+      isExpanded: Boolean,
+      officeHoursOpen: false,
+      disabled: true,
+      selectedEvent: null,
+      isDoubleClick: false,
+      description: '',
+      modifiedTimeStamp: null,
+      loading: true,
+      start:'08:00',
+      end:'17:00',
+      resourceId: 0,
+    };
+    this.scheduleControlInfo = null;
```

Comment:
Is there a reason that this isn't a part of state?

Reply:
Because it contains data from backend and if we use setstate function, the promise order will be confused.